apps/epl/src/epl_timeline.erl

```diff
@@ -0,0 +1,109 @@
+%%% Copyright (c) 2017, erlang.pl
+%%%-------------------------------------------------------------------
+%%% @doc
+%%% Tracking all timeline observers
+%%% @end
+%%%-------------------------------------------------------------------
+-module(epl_timeline).
+-behaviour(gen_server).
+
+%% API
+-export([start_link/0,
+         subscribe/0,
+         unsubscribe/0,
+         handle_pid/2]).
+
+%% gen_server callbacks
+-export([init/1,
+         handle_call/3,
+         handle_cast/2,
+         handle_info/2,
+         terminate/2,
+         code_change/3]).
+
+-record(state, {subscribers = [],
+                timelines = []}).
+
+%%%===================================================================
+%%% API functions
+%%%===================================================================
+
+start_link() ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+subscribe() ->
+    gen_server:cast(?MODULE, {subscribe, self()}).
+
+unsubscribe() ->
+    gen_server:cast(?MODULE, {unsubscribe, self()}).
+
+handle_pid(Action, P) when is_list(P) ->
+    try list_to_pid(P) of
+        Pid -> gen_server:cast(?MODULE, {Action, Pid})
+    catch
+        error:badarg -> noop
+    end;
+
+handle_pid(Action, Pid) when is_binary(Pid) ->
+    handle_pid(Action, binary_to_list(Pid)).
+
+%%%===================================================================
+%%% gen_server callbacks
+%%%===================================================================
+init([]) ->
+    ok = epl:subscribe(),
+    {ok, #state{}}.
+
+handle_call(_Request, _From, State) ->
+    {reply, ok, State}.
+
+handle_cast({subscribe, Pid}, State = #state{subscribers = Subs}) ->
+    {noreply, State#state{subscribers = [Pid|Subs]}};
+handle_cast({unsubscribe, Pid}, State = #state{subscribers = Subs}) ->
+    NextSubs = lists:delete(Pid, Subs),
+    {noreply, State#state{subscribers = NextSubs}};
+handle_cast({add, Pid}, State = #state{timelines = Timelines}) when is_pid(Pid) ->
+    T = case lists:any(fun({P, _}) -> P == Pid end, Timelines) of
+        true -> Timelines;
+        _    ->
+                {ok, Timeline} = epl_timeline_observer:start_link(Pid),
```

Comment:
I've create `epl_timeline` to wrap multiple `epl_timeline_ovserver`s. It was done because observer is able only to track one pid at the time, so because of my lack of Erlang experience I've created wrapper instead refactoring observer structure and interface. We could merge `epl_timeline` with `epl_timeline_observer`